src/Database/LSMTree/Internal/Paths.hs

```diff
@@ -322,16 +320,16 @@ writeBufferFilePathWithExt (WriteBufferFsPaths dir n) ext =
 -------------------------------------------------------------------------------}
 
 toChecksumsFileForWriteBufferFiles :: (ForKOps CRC.CRC32C, ForBlob CRC.CRC32C) -> CRC.ChecksumsFile
-toChecksumsFileForWriteBufferFiles checksums =
-  Map.fromList . toList $ checksums & bimap
```

Comment:
Ah, the `toList (x,y) = [y]`, was that the problem?

Reply:
Yes, indeed. For a foggy moment, my brain believed that the traversable instance of homogeneous tuples would do the right thing.